### DISCUSSION TITLE
Require `override` for callbacks?

### DISCUSSION THREAD

**Discussion by gregmaletic** [finnvoor/PlaydateKit#107] [Ideas]
Very naive question here, and sort of a philosophical one. Would it make sense for `PlaydateGame` callbacks (`update()`, `gameWillPause()`, etc.) to have nil implementations provided in `PlaydateGame`, thereby requiring the a dev to use an `override` keyword to implement them?

The benefits are largely the ones that that come from the `override` keyword in other contexts:

1. Clarity and intent: these methods are not just _any_ method. They purposely need to conform to a specified signature.
2. Compiler safety: it becomes impossible to misspell the method names, or incorrectly specify their signatures.

**Comment by gregmaletic** [comment]
I should have looked at the implementation first: `PlaydateGame` is a protocol extension, not a class. It does define these callbacks, though the `override` keyword isn't needed for the developer's implementation.

I still kinda like having a keyword to override these implementations! But I get the possible benefits for making `PlaydateGame` a protocol.

**Comment by finnvoor** [comment]
I think this is a good point, I don't think there was any specific reason I chose a protocol over an open class. An open class is probably slightly more idiomatic in this case, and I like the benefit of `override` and also allowing you to instantiate an "empty" game.
